src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs

```diff
@@ -211,23 +211,25 @@ private bool IsPotentialPatternVariableDeclaration(SyntaxToken token)
                 return false;
             }
 
-            // Statements containing 'not' cannot be valid variable declarations, e.g. 'e is not int $$' and 'e is not (1 and int $$)'
-            if (patternSyntax.GetAncestorsOrThis(a => a is UnaryPatternSyntax unaryPatternSyntax &&
-                                                      unaryPatternSyntax.PatternOperator.IsKind(SyntaxKind.NotKeyword)).Any())
+            var possibleInvalidVariable = patternSyntax.Parent;
```

Comment:
`for (var current = patternSyntax; current != null; current = current.Parent as PatternSyntax)`